rustls/src/quic.rs

```diff
@@ -130,27 +222,25 @@ pub struct Keys {
 
 impl Keys {
     /// Construct keys for use with initial packets
-    pub fn initial(
-        initial_salt: &hkdf::Salt,
-        client_dst_connection_id: &[u8],
-        is_client: bool,
-    ) -> Self {
+    pub fn initial(initial_salt: &[u8], client_dst_connection_id: &[u8], is_client: bool) -> Self {
         const CLIENT_LABEL: &[u8] = b"client in";
         const SERVER_LABEL: &[u8] = b"server in";
-        let hs_secret = initial_salt.extract(client_dst_connection_id);
+        let hs_secret =
+            hkdf::Salt::new(hkdf::HKDF_SHA256, initial_salt).extract(client_dst_connection_id);
 
         let secrets = Secrets {
             client: hkdf_expand(&hs_secret, hkdf::HKDF_SHA256, CLIENT_LABEL, &[]),
             server: hkdf_expand(&hs_secret, hkdf::HKDF_SHA256, SERVER_LABEL, &[]),
+            suite: TLS13_AES_128_GCM_SHA256_INTERNAL,
         };
-        Self::new(TLS13_AES_128_GCM_SHA256_INTERNAL, is_client, &secrets)
+        Self::new(&secrets, is_client)
```

Comment:
Shouldn't `is_client` be moved into a field of `Secrets` too? That would make `Secrets` harder to misuse because it wouldn't take `is_client` as a parameter.

Reply:
Makes sense, done.